Transaction state fix

gemdev111 opened the pull request:
Done #10
attached: SE3
![Simulator Screenshot - iPhone SE (3rd generation) - 2024-06-06 at 15 11 12](https://github.com/gemwalletcom/gem-ios/assets/171273137/d871faf8-9e5e-4220-9e6f-5e5e0b3f0533)

gemdev111 committed:
Refactored ListItemView, added previews with different states
moved ListItemVIew to Lists folder

gemdev111 committed:
moved ListItemProgressView from ListItemView, since it has more then 1 dependecy.  Also used in TransactionsScene separetly

gemdev111 committed:
Project strucuture update
TextStyle moved from Components package to Style package, added necessary Style package imports to Gem
ListItemView minor change

Done #39

gemdev111 committed:
Merge branch 'main'

gemdev111 committed:
fixed pbroj, unlinked files after vsc

gemdev111 committed:
Added simple previews to TransactionView
Fixing preview crashing by updating AssetId.binance in App-Configuration

gemdev111 committed:
Fixed: #10 Transaction state does not always fit

gemcoder21 commented on Packages/Components/Sources/Lists/ListItemView.swift:
```diff
@@ -0,0 +1,392 @@
+
+// TODO: - disscuss and maybe move this later somewhere
+
+public struct ListItemValue<T> {
```
move to another file`ListItemValue`, here

gemcoder21 commented on Packages/Components/Sources/Lists/ListItemView.swift:
```diff
@@ -0,0 +1,392 @@
+            )
+        }
+
+        // Long Text States
```
Remove `// Long Text States`, it says it in the section name

gemcoder21 commented on Packages/Components/Sources/Lists/ListItemView.swift:
```diff
@@ -0,0 +1,392 @@
+    let tagTextStyleBlue = TextStyle(font: Font.system(.footnote), color: .blue, background: .blue.opacity(0.2))
+
+    return List {
+        // Basic States
```
```suggestion
```

gemcoder21 commented on Packages/Components/Sources/Lists/ListItemView.swift:
```diff
@@ -0,0 +1,392 @@
+            )
+        }
+
+        // Image States
```
```suggestion
```

gemcoder21 commented on Packages/Components/Sources/Lists/ListItemView.swift:
```diff
@@ -0,0 +1,392 @@
+            )
+        }
+
+        // Combined States
```
```suggestion
```

gemcoder21 commented on Packages/Components/Sources/Lists/ListItemView.swift:
```diff
@@ -0,0 +1,392 @@
+            )
+        }
+
+        // Additional Scenarios
```
```suggestion
```

gemdev111 committed:
Removed unused comments
ListItemValue moved to Lists folder as seprate file
ListSection moved to Lists folder
